### PR TITLE
No sandbox data segments

### PR DESCRIPTION
**Pull request by matthias-blume**
First cut of the no-sandboxing of data segments. 

This embeds un-sandboxed function- and data pointers into the initializer.
Each data segment that contains such pointers is rendered as a custom
struct with pointer elements (represented as unsigned integers) inserted
at the right places.

### CODE REVIEW

File: src/c-writer.cc [hunk 1]
```diff
@@ -3968,6 +4124,35 @@ void CWriter::Write(const ConvertExpr& expr) {
   }
 }
 
+void CWriter::WriteFunctionAddress(Index symbol_index, bool is64bit) {
+  Index func_index = module_->function_symbols_.at(symbol_index);
+  const std::string& func_name = module_->funcs[func_index]->name;
+  Write("reinterpret_cast<u", is64bit ? "64" : "32", ">(&",
```
Comment (dimitry-) [inline src/c-writer.cc]: afaik C does not have reinterpret_cast - the only available kind is the C-cast. Sorry I didn't catch it before.

Comment (matthias-blume) [reply src/c-writer.cc]: Good point.  But that's an easy fix.  Thanks!

**Review by dimitry- — APPROVED** [review]
otherwise lgtm

File: src/c-writer.cc (hunk 1 above)
Comment (matthias-blume) [reply src/c-writer.cc]: Done.

**Pull request merged**